lib/resources.js

```diff
@@ -657,9 +675,15 @@ module.exports = function (app, _private = null) {
    * @returns {object}
    */
   const itemsQueryContext = (options) => {
-    return options.merge_checkin_card_items
-      ? { must: { match_all: {} } }
-      : { must_not: { term: { 'items.type': 'nypl:CheckinCardItem' } } }
+    if (options.merge_checkin_card_items && options.removeElectronicResourcesFromItemsArray) {
+      return { must_not: { exists: { field: 'items.electronicLocator' } } }
+    }
+
+    if (options.merge_checkin_card_items) {
+      return { must: { match_all: {} } }
+    }
+
+    return { must_not: [{ term: { 'items.type': 'nypl:CheckinCardItem' } }, { exists: { field: 'items.electronicLocator' } }] }
```

Comment:
This reads to me as:
```
If (options.merge_checkin_card_items && options.removeElectronicResourcesFromItemsArray):
  - exclude items with electronicLocator, include checkin-cards
else if (options.merge_checkin_card_items):
  - include all (checkin-cards and items with electronicLocator)
else:
  - exclude checkin-card items and include items with electronicLocator
```

You have two options, so there are four possibilities, but I think you're only handling three:
 - include checkin-cards, exclude e-resources
 - include checkin cards, include e-resources
 - exclude checkin cards, exclude e-resources

Since the options ultimately just control what's in the `must_not`, is it possible to structure it like this:
```
const excludeClauses = []

if (options.removeElectronicResourcesFromItemsArray)
  excludeClauses.push({ exists: { field: 'items.electronicLocator' } })
  
if (!options.merge_checkin_card_items)
  excludeClauses.push({ term: { 'items.type': 'nypl:CheckinCardItem' } })

if (excludeClauses.length) return { must_not: excludeClauses }
else return { must: { match_all: {} } }
```

Reply:
Good point, I've incorporated that change